runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -130,7 +136,7 @@
      */
     @Deprecated // TODO: Add replacement method that conforms to style
     @SuppressWarnings("checkstyle:abbreviation") // Due to deprecation
-    public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[] args,
+    public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, String streamName, Class<T> type, Object[] args,
```

Comment:
streamName and streamId are redundant information, I don't think its good to pass them around separately. Keeping the name and computing a UUID should be cleaner, since the converse isn't true (i.e can't map to String from UUID).

Reply:
Done